CMakeLists.txt

```diff
@@ -0,0 +1,1045 @@
+
+	if(STATIC)
+		SET(CMAKE_CXX_LINKER_FLAGS "-static ${CMAKE_CXX_LINKER_FLAGS}")
+	endif(STATIC)
+
+endif()
+
+#MPI default build options
+SET( CMAKE_CXX_FLAGS_MPI "-DNDEBUG -Wall -O3 -pedantic -DENABLE_MPI -march=native" CACHE STRING
+	"Flags used by the C++ compiler during MPI builds."
+	FORCE )
+SET( CMAKE_EXE_LINKER_FLAGS_MPI
+	"-Wl,--warn-unresolved-symbols,--warn-once" CACHE STRING
+	"Flags used for linking binaries during MPI builds."
+	FORCE )
+SET( CMAKE_SHARED_LINKER_FLAGS_MPI
+	"-Wl,--warn-unresolved-symbols,--warn-once -fPIC -flto" CACHE STRING
+	"Flags used by the shared libraries linker during MPI builds."
+	FORCE )
+MARK_AS_ADVANCED(
+	CMAKE_CXX_FLAGS_MPI
+	#CMAKE_C_FLAGS_MPI
+	CMAKE_EXE_LINKER_FLAGS_MPI
+	CMAKE_SHARED_LINKER_FLAGS_MPI)
+
+#MPI debug default build options
+SET( CMAKE_CXX_FLAGS_MPIDEBUG "-Wall -g3 -ggdb -pedantic -DENABLE_MPI -march=native" CACHE STRING
+	"Flags used by the C++ compiler during MPI Debug builds."
+	FORCE )
+SET( CMAKE_EXE_LINKER_FLAGS_MPIDEBUG
+	"-Wl,--warn-unresolved-symbols,--warn-once" CACHE STRING
+	"Flags used for linking binaries during MPI Debug builds."
+	FORCE )
+SET( CMAKE_SHARED_LINKER_FLAGS_MPIDEBUG
+	"-Wl,--warn-unresolved-symbols,--warn-once -fPIC" CACHE STRING
+	"Flags used by the shared libraries linker during MPI Debug builds."
+	FORCE )
+MARK_AS_ADVANCED(
+	CMAKE_CXX_FLAGS_MPIDEBUG
+	#CMAKE_C_FLAGS_MPIDEBUG
+	CMAKE_EXE_LINKER_FLAGS_MPIDEBUG
+	CMAKE_SHARED_LINKER_FLAGS_MPIDEBUG )
+
+###################################################################################################
+# Settings for what exactly to build
+###################################################################################################
+#add the source files that will be used to build sixtrack
+add_executable(SixTrack ${SIXTRACK_SOURCES})
+SET_TARGET_PROPERTIES(SixTrack PROPERTIES OUTPUT_NAME "${SIXTRACK_BINARY_NAME}" LINK_FLAGS "${CMAKE_Fortran_LINKER_FLAGS}")
+
+#build astuce
+add_executable(astuce ${CMAKE_SOURCE_DIR}/SixTrack/astuce.f)
+SET_TARGET_PROPERTIES(astuce PROPERTIES LINK_FLAGS "${CMAKE_Fortran_LINKER_FLAGS}")
+if(Fortran_COMPILER_NAME MATCHES "nagfor.*")
+	target_link_libraries(astuce ${g2cLink})
+endif()
+
+#build dafor
+add_executable(dafor ${CMAKE_SOURCE_DIR}/SixTrack/dafor.f)
+SET_TARGET_PROPERTIES(dafor PROPERTIES LINK_FLAGS "${CMAKE_Fortran_LINKER_FLAGS}")
+
+if(CRLIBM)
+	file(GLOB CRLIBM_SOURCES ${CMAKE_SOURCE_DIR}/SixTrack/crlibm/*.c)
+	add_library(crlibm STATIC ${CRLIBM_SOURCES})
+
+	target_compile_definitions(crlibm PRIVATE -DLINUX_INLINE -DHAVE_CONFIG_H)
+
+	if(32BIT)
+		target_compile_options(crlibm PRIVATE -m32)
+	endif(32BIT)
+	if(64BIT)
+		target_compile_options(crlibm PRIVATE -m64)
+	endif(64BIT)
+
+	if(${CMAKE_C_COMPILER_ID} MATCHES "GNU")
+		target_compile_options(crlibm PRIVATE -fPIC -std=c99 -Wall -Wshadow -Wpointer-arith -Wcast-align -Wconversion -Waggregate-return -Wstrict-prototypes -Wnested-externs -Wlong-long -Winline -pedantic -fno-strict-aliasing)
```

Comment:
Great! Can't see that splitup yet tough, did you push it?

> Do we ever still want to deal with ARM based computers?

I know Eric is keen to empty my phone battery. But let's get this thing working on x86 first...